lib/archethic/utils.ex

```diff
@@ -913,4 +916,62 @@ defmodule Archethic.Utils do
     |> Stream.take_while(&(NaiveDateTime.compare(&1, end_of_month_datetime) in [:lt]))
     |> Enum.count()
   end
+
+  @doc """
+  get token properties based on the genesis address and the transaction
+  """
+  @spec get_token_properties(binary(), Transaction.t()) :: {:ok, map()} | {:error, atom()}
+  def get_token_properties(genesis_address, %Transaction{
+        data: %TransactionData{
+          content: content,
+          ownerships: ownerships
+        },
+        type: type
+      })
+      when type in [:token, :mint_rewards] do
+    case Jason.decode(content) do
+      {:ok, map} ->
+        result = %{
+          genesis: genesis_address,
+          name: Map.get(map, "name", ""),
+          supply: Map.get(map, "supply"),
+          symbol: Map.get(map, "symbol", ""),
+          type: type,
```

Comment:
The type should be retrieve from the map, the expected type are `fungible` or `non-fungible` and not the transaction type

Reply:
done